platform/tests/UNITTESTS/doubles/CMakeLists.txt

```diff
@@ -17,14 +26,17 @@ target_sources(mbed-stubs-platform
         randLIB_stub.c
         randLIB_stub.cpp
 )
-
+target_include_directories(mbed-stubs-platform
+    PUBLIC
```

Comment:
Instead of adding all the platform stub headers into mbed-stubs-platform with PUBLIC  again, this will be a problem if library unittest (example ATCmdParser unittest) don't want to include stub platform headers but target link aginst mbed-stubs-platform brought those the platform stub headers. Initially, that was one of the reasons we added a mbed-stubs-headers library, it is better to add a new library mbed-stubs-platform-headers INTERFACE library with all platform stub headers refer our discussion https://github.com/ARMmbed/mbed-os/pull/14285#discussion_r576779176

Reply:
I'm not sure I'm seeing the issue with the example use case you've given. Why wouldn't you want the headers to be available if you specifically linked to the stub library? I could see the inverse of this being a potential problem: you want the "stub header" but want to provide another implementation yourself. However, if a stub has some "public headers" (every header in this `mbed-stubs-headers` lib is "public") it seems better to expose them as part of the stub library they belong to. Pulling all the headers out into a monolithic header-only library that requires non-obvious dependencies before you can use it seems like an anti-pattern.

Maybe I'm missing something, so if you could show me an example of where this is actually an issue it would help me understand why we need to do this. It doesn't seem to cause a problem with the `ATCmdParser` test you mentioned.

Reply:
In the original discussion @rajkan01 linked, @paul-szczepanek-arm said

> If they are PUBLIC then if you depend on this library you also depend on the libraries it depends on.
>
> This is like a header relying on another header to include something rather than including it itself. I think it's better if things are self contained and just exposing thing they are meant to expose and not what they are using themselves.

My interpretation of the concern is, if stub libraries A and B both have PUBLIC headers, and B depends on A, and a test T directly depends on B, then A will be "inadvertently" be exposed to A, is this the concern? But in order for compilation to succeed (because of how preprocessors work), this exposure is necessary IMO - otherwise how can T `#include` something from B which then `#include` something from A? Both A and B need to be in the include path.

Another thing is, the stub headers are for unit tests only which works within mbed-os, not intended for general use by other repositories, so flexibility isn't a concern and we should prioritise simplicity in my opinion.

(Sorry if I completely misunderstood the original concern!)